wip-0002.md

```diff
@@ -0,0 +1,212 @@
+
+**All committers must be listed.** All identities that produced commitment transactions for a data request that exist in the block chain prior to the tally transaction itself ("committers") must be present in either one of the `outputs` or `slashed_witnessess` fields.
+
+**No identity repetition.** No identity can appear more than once in either the `output` or `slashed_identities` fields. In the same manner, no identity can be present in both the `outputs` and `slashed_identities` fields, with the only exception of the requestor, which in addition to being in one of those fields, can calso be present in (a) both at the same time OR (b) twice in `outputs` (see *"Change output"* below).
+
+**Requests with insufficient commits**. Given a data request, may not exist enough commitments for it in the `extra_commit_rounds + 1` number of blocks (as specified by the request) following the block in which it was bound to enter the commit stage, a tally can be immediately produced such that the result is the "Insufficient commits" RADON error, the `slashed_identities` field is empty, and the `outputs` field contains only the change output (see *"Change output"* below).
+
+**Missing reveals**. All committers for which no valid reveal transaction exists in the block chain prior to the tally transaction itself must be listed in the`slashed_witnessess` field. To this effect, for a reveal transaction to be considered valid, it must be properly constructed such that it can be decoded into a compliant RADON type or RADON error.
+
+**Tally evaluation.** The rules for deciding whether committers which have produced valid reveal transactions for the referred request such that they exist prior to the tally transaction itself (revealers) belong to the `output` or `slashed_identities` fields remain the same as they were before this proposal.
+
+**Change output.** Given a request such that (1) it had insufficient commits, or (2) its count of committers outnumbers that of revealers, or (3) its count of slashed identies is not zero, or (4) any combination of 1, 2 and 3, a first `ValueTransferOutput` entry (change output) must exist in `outputs` such that it is spendable by the requestor, and its value equals the total value that was expected to be used for the `commit_fee` of the missing commits, plus the total value that was expected to be used for the `reward` and `reveal_fee` of the missing reveals, plus all the value that that was expected to be used as the `reward` of the slashed identities:
+
+    change = (reward + commit_fee) * (witnesses - committers_count) +
+             (reward + reveal_fee) * (witnesses - revealers_count) +
+             reward * slashed_count
```

Comment:
* (witnesses - revealers_count) = non revealers, this guys are included also in slashed_count so you can not include the reward two times
* non- committers are also non revealers so... same comment

Another possible formula could be:
    change = reward * (witnesses - honests_count)
        + reveal_fee * (witnesses - reveals_count)
        + commit_fee * (witnesses - commits_count)
(Where honest_ count are node that produces a reveal that is in consensus with the majority)

Reply:
What about this:
```
    change = (reward + commit_fee) * (witnesses - committers_count) +
             reveal_fee * (committers_count - revealers_count) +
             reward * slashed_count
```

Reply:
I mean, it is the same as you posted but using `slashed_count` instead of `witnesses - honests_count` so as not to introduce another variable (`honest_count`) and expressed chronologically (commitment first, then reveal, and finally the slashing).

Reply:
in case of Insufficient commits error reveal_fee has to be returned to the data requester